skyportal/models.py

```diff
@@ -2501,6 +2517,49 @@ class FacilityTransaction(Base):
 )
 
 
+class Listing(Base):
+
+    user_id = sa.Column(
+        sa.ForeignKey('users.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="The ID of the User who created this Listing.",
+    )
+
+    user = relationship(
+        "User",
+        foreign_keys=user_id,
+        back_populates="listings",
+        doc="The user that saved this object/listing",
+    )
+
+    obj_id = sa.Column(
+        sa.ForeignKey('objs.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="The ID of the object that is on this Listing",
+    )
+
+    obj = relationship(
+        "Obj", back_populates="listings", doc="The object referenced by this listing",
+    )
+
+    list_name = sa.Column(
+        sa.String, index=True, doc="Name of the list, e.g., 'favorites'. ",
+    )
+
+
+Listing.__table_args__ = (
+    sa.Index(
+        "listings_main_index",
+        Listing.user_id,
+        Listing.obj_id,
+        Listing.list_name,
+        unique=True,
+    ),
+)
```

Comment:
Add a reverse index, 

```
Listing.__table_args__ = (
    sa.Index(
        "listings_main_index",
        Listing.obj_id,
        Listing.user_id,
        Listing.list_name,
        unique=True,
    ),
)

Reply:
I don't know what that means...

Reply:
In a table with 2 columns A, B an index on (A, B) != an index on (B, A). Some queries need one and not the other, so I am suggesting to add an index on (B, A, C) in addition to (A, B, C) which you have already